Firestore/core/test/firebase/firestore/testutil/testutil.h

```diff
@@ -27,6 +27,8 @@
 #include <vector>
 
 #include "Firestore/core/include/firebase/firestore/timestamp.h"
+#include "Firestore/core/src/firebase/firestore/api/document_reference.h"
+#include "Firestore/core/src/firebase/firestore/api/firestore.h"
```

Comment:
I don't think the api layer will work in straight c++. You may need to protect these includes (and tests that require them) with `ifdef __OBJC__` or similar.

Reply:
The API layer explicitly _does_ work in straight C++ (and boy did I have to work to make it so--see `objc_class.h`). See `cc_compilation_test.cc` that verifies exactly this property.

Reply:
Well, it doesn't work with *my* compiler. :)

There's a bunch of objc-isms in the headers that cause breakage. eg NS_ASSUME_NONNULL_BEGIN, etc. You mentioned in another comment that you were going to look at the non-objc breakages separately, so feel free to defer this until then.